restreamer/restreamer/main.py

```diff
@@ -163,6 +168,96 @@ def generate_media_playlist(stream, variant):
 	return generate_hls.generate_media(segments, os.path.join(app.static_url_path, stream, variant))
 
 
+@app.route('/cut/<stream>/<variant>.ts')
+@has_path_args
+def cut(stream, variant):
+	"""Return a MPEGTS video file covering the exact timestamp range.
+	Params:
+		start, end: Required. The start and end times, down to the millisecond.
+			Must be in ISO 8601 format (ie. yyyy-mm-ddTHH:MM:SS).
+		allow_holes: Optional, default false. If false, errors out with a 406 Not Acceptable
+			if any holes are detected, rather than producing a video with missing parts.
+			Set to true by passing "true" (case insensitive).
+			Even if holes are allowed, a 406 may result if the resulting video would be empty.
+	"""
+	start = dateutil.parser.parse(request.args['start'])
+	end = dateutil.parser.parse(request.args['end'])
+	if end <= start:
+		return "End must be after start", 400
+
+	allow_holes = request.args.get('allow_holes', 'false').lower()
+	if allow_holes not in ["true", "false"]:
+		return "allow_holes must be one of: true, false", 400
+	allow_holes = allow_holes == "true"
```

Comment:
I had to reread this a couple times to make sure it wasn't a coding error. I'd toss a pair of parentheses around the comparison to make it more clear that you are assigning the result of the comparison.